### PR TITLE
chore: update typescript, optimize types

### PR DESCRIPTION
**Pull request by LufyCZ**
## PR-Codex overview
This PR updates the TypeScript version across multiple packages for enhanced typechecking and performance.

### Detailed summary
- Updated TypeScript version from 5.4.5 to 5.5.4 in various package.json files
- Added typecheck speedup optimizations in multiple files for better performance

> The following files were skipped due to too many changes: `pnpm-lock.yaml`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: apis/extractor/src/request-statistics.ts [hunk 1]
```diff
@@ -150,7 +150,7 @@ export class RequestStatistics {
 
   requestRejected(reason: ResponseRejectReason) {
     ++this.rejected
-    ++this.rejects[reason]
+    this.rejects[reason] = (this.rejects[reason] || 0) + 1
```
Comment (matthewlilley) [inline apis/extractor/src/request-statistics.ts:153]: all reasons are set to 0 already in constructor so this seems totally uneccasary

File: apis/router/src/SwapRequestStatistics.ts [hunk 2]
```diff
@@ -160,7 +160,7 @@ export class SwapRequestStatistics {
 
   requestRejected(reason: ResponseRejectReason) {
     ++this.rejected
-    ++this.rejects[reason]
+    this.rejects[reason] = (this.rejects[reason] || 0) + 1
```
Comment (matthewlilley) [inline apis/router/src/SwapRequestStatistics.ts:163]: all reasons are set to 0 already in constructor so this seems totally uneccasary

File: apis/router/test/pricesCompare.ts [hunk 3]
```diff
@@ -31,7 +31,7 @@ export function comparePrices(
         totalR += r
         maxR = Math.max(r, maxR)
         levels.forEach((l, i) => {
-          if (r > l) levelsEx[i] += 1
+          if (r > l) levelsEx[i] = (levelsEx[i] || 0) + 1
```
Comment (matthewlilley) [inline apis/router/test/pricesCompare.ts:34]: we know this is set to 0 already "const levelsEx = levels.map((_) => 0)"

File: apis/router/src/prices.ts [hunk 4]
```diff
@@ -241,7 +241,7 @@ export function comparePrices(
         totalR += r
         maxR = Math.max(r, maxR)
         levels.forEach((l, i) => {
-          if (r > l) levelsEx[i] += 1
+          if (r > l) levelsEx[i] = (levelsEx[i] || 0) + 1
```
Comment (matthewlilley) [inline apis/router/src/prices.ts:244]: we know this is set to 0 already "const levelsEx = levels.map((_) => 0)"

File: apps/web/src/ui/swap/cross-chain/cross-chain-swap-trade-review-dialog.tsx [hunk 5]
```diff
@@ -89,6 +92,28 @@ import {
...
 
+function getConfig(trade: UseCrossChainTradeReturn | undefined) {
+  if (!trade) return {}
+
+  if (trade.functionName === SushiXSwapFunctionName.Bridge) {
+    return {
+      abi: sushiXSwap2Abi_swap,
+      functionName: 'swap',
+      args: trade.writeArgs as NonNullable<SushiXSwapWriteArgsBridge>,
+      value: BigInt(trade.value ?? 0) as any,
```
Comment (matthewlilley) [inline apps/web/src/ui/swap/cross-chain/cross-chain-swap-trade-review-dialog.tsx:103]: cant these br undefined?

File: apps/web/src/ui/swap/cross-chain/cross-chain-swap-trade-review-dialog.tsx [hunk 6]
```diff
@@ -89,6 +92,28 @@ import {
...
+    } as const
+  }
+
+  if (trade.functionName === SushiXSwapFunctionName.SwapAndBridge) {
+    return {
+      abi: sushiXSwap2Abi_swapAndBridge,
+      functionName: 'swapAndBridge',
+      args: trade.writeArgs as NonNullable<SushiXSwapWriteArgsSwapAndBridge>,
+      value: BigInt(trade.value ?? 0) as any,
```
Comment (matthewlilley) [inline apps/web/src/ui/swap/cross-chain/cross-chain-swap-trade-review-dialog.tsx:112]: cant these br undefined?

File: apps/web/src/ui/swap/cross-chain/cross-chain-swap-trade-review-dialog.tsx [hunk 7]
```diff
@@ -315,8 +337,11 @@ export const CrossChainSwapTradeReviewDialog: FC<{ children: ReactNode }> = ({
     },
   })
 
+  // Speeds up typechecking in the useMemo below
+  const _simulation: { request: any } | undefined = simulation
+
   const write = useMemo(() => {
-    if (!simulation) return undefined
+    if (!_simulation?.request) return undefined
```
Comment (matthewlilley) [inline apps/web/src/ui/swap/cross-chain/cross-chain-swap-trade-review-dialog.tsx:344]: is this a better check?

File: apps/web/turbo.json [hunk 8]
```diff
@@ -13,7 +13,8 @@
         ".env"
       ],
       "outputs": [".next/**", "!.next/cache/**"],
-      "dependsOn": ["^build"]
+      "dependsOn": ["^build"],
+      "env": ["NEXT_PUBLIC_APP_ENV", "EDGE_CONFIG"]
```
Comment (matthewlilley) [inline apps/web/turbo.json:17]: Should these affect the build hashes?

File: apis/extractor/src/request-statistics.ts (hunk 1 above)
Comment (LufyCZ) [reply apis/extractor/src/request-statistics.ts:153]: doesn't really matter as long as ts is happy

File: apis/router/src/SwapRequestStatistics.ts (hunk 2 above)
Comment (LufyCZ) [reply apis/router/src/SwapRequestStatistics.ts:163]: doesn't really matter as long as ts is happy

File: apis/router/src/prices.ts (hunk 4 above)
Comment (LufyCZ) [reply apis/router/src/prices.ts:244]: doesn't really matter as long as ts is happy

File: apps/web/src/ui/swap/cross-chain/cross-chain-swap-trade-review-dialog.tsx (hunk 6 above)
Comment (LufyCZ) [reply apps/web/src/ui/swap/cross-chain/cross-chain-swap-trade-review-dialog.tsx:112]: don't think so. as long as trade is defined, writeArgs is, too

File: apps/web/src/ui/swap/cross-chain/cross-chain-swap-trade-review-dialog.tsx (hunk 7 above)
Comment (LufyCZ) [reply apps/web/src/ui/swap/cross-chain/cross-chain-swap-trade-review-dialog.tsx:344]: it's pretty much the same, was just bringing it in line with other files where it's simulation?.request

File: apps/web/turbo.json (hunk 8 above)
Comment (LufyCZ) [reply apps/web/turbo.json:17]: think we used to have issues with those. not sure why it's in this pr though

Comment (LufyCZ) [reply apps/web/turbo.json:17]: we need to either disable the lint rule causing all envs to be errors if they aren't in turbo.json or add all the envs in there

File: packages/react-query/src/hooks/trade/types.ts [hunk 9]
```diff
@@ -1,11 +1,14 @@
-import { routeProcessor2Abi } from 'sushi/abi'
 import { ChainId } from 'sushi/chain'
 import { Amount, Price, type Type } from 'sushi/currency'
 import { Percent } from 'sushi/math'
 import { RouterLiquiditySource } from 'sushi/router'
 import type { Address, WriteContractParameters } from 'viem'
 import z from 'zod'
 
+import {
+  routeProcessor2Abi_processRoute,
```
[the remark this answers is not in the document]
Comment (matthewlilley) [reply packages/react-query/src/hooks/trade/types.ts:9]: maybe use v5 and delete everything prior to v4

File: apps/web/src/ui/swap/cross-chain/cross-chain-swap-trade-review-dialog.tsx (hunk 6 above)
Comment (matthewlilley) [reply apps/web/src/ui/swap/cross-chain/cross-chain-swap-trade-review-dialog.tsx:112]: I mean the value: undefined, and the as any

https://wagmi.sh/core/api/actions/call#value

**Pull request merged**